core/base/src/TDirectory.cxx

```diff
@@ -96,8 +96,12 @@ TDirectory::TDirectory(const char *name, const char *title, Option_t * /*classna
 
 TDirectory::~TDirectory()
 {
-   if (!gROOT) {
-      delete fList;
+   // Use gROOTLocal to avoid triggering undesired initialization of gROOT.
+   // For example in compiled C++ programs that don't use it directly.
+   if (!ROOT::Internal::gROOTLocal) {
+      if (fList) {
+         delete fList;
+      }
```

Comment:
```suggestion
   if (!ROOT::Internal::gROOTLocal) {
         delete fList;
```

Reply:
Ok, then I assume there's no way on Earth or otherwise that `fList==nullptr` here. I will trust you on this.